lib/rack/attack.rb

```diff
@@ -146,6 +146,9 @@ def clear!
       retry_after = (env['rack.attack.match_data'] || {})[:period]
       [429, { 'Content-Type' => 'text/plain', 'Retry-After' => retry_after.to_s }, ["Retry later\n"]]
     end
+    @discriminator_normalizer = lambda do |discriminator|
+      discriminator.to_s.downcase.chomp.gsub(/\s+/, "")
```

Comment:
Consider checking if the `.chomp` call is still needed, it may be covered by the `gsub` call.

Reply:
Still needed, because it removes `\r` and `\n` while regex removes only all kinds of whitespaces

Reply:
Try this in `irb`:

```
irb(main):014:0> "hello\r\n\r\r\n".strip
=> "hello"
irb(main):015:0> "hello\r\n\r\r\n".chomp
=> "hello\r\n\r"
irb(main):016:0> "hello\r\n\r\r\n".gsub(/\s+/, "")
=> "hello"
irb(main):017:0> 
```

Reply:
Ah, my bad. I used single quotes instead of double quotes for string. Nice catch! Updated.